content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md

```diff
@@ -394,3 +394,17 @@ For more details on etcd upgrade, please refer to the [etcd upgrades](https://et
 {{< note >}}
 Before you start an upgrade, please back up your etcd cluster first.
 {{< /note >}}
+
+## Maintaining etcd clusters
+
+Fore more details on etcd maintenance, please refer to the [etcd maintenance](https://etcd.io/docs/latest/op-guide/maintenance/) documentation.
+
+{{% thirdparty-content %}}
+
+{{< note >}}
+Defragmentation is an expensive operation, so it should be executed as infrequent
+as possible. On the other hand, it's also necessary to make sure any etcd member
+will not run out of the storage quota. It's recommended to run defragmentation
+using [etcd-defrag](https://github.com/ahrtr/etcd-defrag), which is an easier to
+use and smarter etcd defragmentation tool.
```

Comment:
```suggestion
will not run out of the storage quota. The Kubernetes project recommends that when you
perform defragmentation, you use a tool such as [etcd-defrag](https://github.com/ahrtr/etcd-defrag).
```
or
```suggestion
will not run out of the storage quota. The Kubernetes project recommends that when you
perform defragmentation, you use a tool such as [etcd-defrag](https://github.com/ahrtr/etcd-defrag).
Compared to the tooling provided by the etcd project itself, `etcd-defrag` is easier to use and smarter.
```

Ideally, avoid this kind of comparative evaluation. We try to avoid making critiques between different third party tools, as these are an extra effort to maintain. It's hard to find Kubernetes contributors who can keep that kind of information current.

Reply:
thx & updated. PTAL